lib/pos_blockchain.go

```diff
@@ -589,6 +633,46 @@ func (bc *Blockchain) hasValidBlockViewPoS(block *MsgDeSoBlock) error {
 	return nil
 }
 
+func (bav *UtxoView) hasValidProposerPartialSignaturePoS(block *MsgDeSoBlock, snapshotAtEpochNumber uint64) (bool, error) {
+	votingPublicKey := block.Header.ProposerVotingPublicKey
+	proposerPublicKey := block.Header.ProposerPublicKey
+	proposerPartialSig := block.Header.ProposerVotePartialSignature
+	// If the proposer partial sig is nil, we can't validate it. That's an error.
+	if proposerPartialSig.IsEmpty() {
+		return false, nil
+	}
+	// Get the snapshot validator entry for the proposer.
+	proposerPKID := bav.GetPKIDForPublicKey(proposerPublicKey.ToBytes()).PKID
+	snapshotValidatorEntry, err := bav.GetSnapshotValidatorSetEntryByPKIDAtSnapshotAtEpochNumber(proposerPKID, snapshotAtEpochNumber)
```

Comment:
Can you filter this to just the leader schedule for the epoch instead? I assume you're intended to make sure the proposer is actually an eligible leader for the epoch. Rough steps to perform this check are:
1. Fetch the entire snapshot leader schedule for the epoch, and find the leader with the above PKID
2. Fetch the snapshot validator corresponding to the leader's PKID
3. Validate the signature

Reply:
This is already caught in the leader check in the function `hasValidBlockProposerPoS`, so this would be duplicative. Basically this function just checks that the `ProposerPublicKey`, `ProposerVotingPublicKey`, and `ProposerVotePartialSignature` are correct. By not forcing this function to validate the leader, we're able to use this check in validating orphans from the previous & next epoch. When we go to process a "former orphan", we'll do the proper leader check and it'll be caught there.

Reply:
We discussed offline that the leader check is performed elsewhere. Marking this as resolved.